src/System.Net.WebSockets.Client/tests/ClientWebSocketTest.cs

```diff
@@ -27,9 +28,46 @@ public ClientWebSocketTest(ITestOutputHelper output)
             _output = output;
         }
 
+        public static IEnumerable<object[]> UnavailableWebSocketServers
+        {
+            get
+            {
+                Uri server;
+                
+                // Unknown server.
+                {
+                    server = new Uri(string.Format("ws://{0}", Guid.NewGuid().ToString()));
+                    yield return new object[] { server };
+                }
+
+                // Known server but not a real websocket endpoint.
+                {
+                    server = HttpTestServers.RemoteEchoServer;
+                    var ub = new UriBuilder("ws", server.Host, server.Port, server.PathAndQuery);
+
+                    yield return new object[] { ub.Uri };
+                }
+            }
+        }
+
         private static bool WebSocketsSupported { get { return WebSocketHelper.WebSocketsSupported; } }
 
 #region Connect
+        [ConditionalTheory("WebSocketsSupported"), MemberData("UnavailableWebSocketServers")]
```

Comment:
These newly added tests in this PR, like the other tests ported to GitHub, use the same common WebSockets.Client functionality. So, they should all pass the same.